File: scripts/python/parsers/profile_parser.py
"""
profile_parser.py
Reads Salesforce Profile metadata from SFDX source format.
Uses stdlib only (pathlib).
"""

from pathlib import Path


def parse_profiles(profile_dir: str) -> dict:
    """
    Parse all profiles under profile_dir.
    Returns {'profiles': [{'name'}]}
    """
    profiles = []
    p = Path(profile_dir)

    if not p.exists():
        return {'profiles': profiles}

    for ff in sorted(p.glob('*.profile-meta.xml')):
        name = ff.name.replace('.profile-meta.xml', '')
        profiles.append({'name': name})

    return {'profiles': profiles}

File: scripts/node/parsers/profileParser.js
/**
 * profileParser.js
 * Reads Salesforce Profile metadata from SFDX source format.
 */

'use strict';

const fs   = require('fs');
const path = require('path');

/**
 * @param {string} profileDir  - path to the profiles/ folder
 * @returns {{ profiles: Array<{ name }> }}
 */
async function parseProfiles(profileDir) {
  const profiles = [];

  if (!fs.existsSync(profileDir)) return { profiles };

  const files = fs.readdirSync(profileDir).filter(f => f.endsWith('.profile-meta.xml'));

  for (const file of files) {
    const name = file.replace('.profile-meta.xml', '');
    profiles.push({ name });
  }

  return { profiles };
}

module.exports = { parseProfiles };
